web_poet/overrides.py

```diff
@@ -141,22 +162,35 @@ def handle_urls(
         """
 
         def wrapper(cls):
-            rule = OverrideRule(
+            final_data_type = to_return
+            derived_type = get_generic_parameter(cls)
+            if derived_type and derived_type != ItemT:
+                final_data_type = derived_type
```

Comment:
I think we have the code which does the same in `Returns.item_cls`. It'd be good to have only one copy. I can see two ways to achieve it:

1) Make Returns.item_cls a class property (could be tricky), or expose another class method for it.
2) Extract this code to an utility function.

Reply:
Exposing it into another classmethod won't be usable for the following use case since it's not inheriting from `Returns`:
```python
@handle_urls("example.com", to_return=Product)
class CustomProductPageDataTypeOnly(web_poet.Injectable):
    ...
```

Assuming that we're able to extract it into another utility function like:
```python
def get_item_cls(cls) -> typing.Type[ItemT]:
    param = get_generic_parameter(cls)
    if isinstance(param, typing.TypeVar):
        return dict
    return param
```

This code-block would still be more or less the same:
```python
final_data_type = to_return
derived_type = get_item_cls(cls)
if derived_type and derived_type != dict:  # the only diff is we're checking against a dict
    final_data_type = derived_type
```

I believe we've already reduced this one to its core.

Reply:
I now see what you mean, thanks for explaining it further! Updated this in https://github.com/scrapinghub/web-poet/pull/84/commits/144e39ef36d5db891686bae8137f723ba4e1437f

> The idea is that to_return passed explicitly to the decorator should take priority over the item class defined by the page object.
I currently had it the other way around. Nice catch!